src/context.rs

```diff
@@ -390,6 +392,7 @@ impl PySessionContext {
         Ok(PyDataFrame::new(df))
     }
 
+    #[pyo3(signature = (partitions, name=None, schema=None))]
```

Comment:
Does pyo3 v0.22 require this? Otherwise we can save some effort and only handle these signatures in the wrapper side

Reply:
I pulled your branch and I see this probably due to the deprecated warning you get without it.

Reply:
Upstream pyo3 change: [deprecate implicit default for trailing optional arguments](https://github.com/PyO3/pyo3/pull/4078)

Reply:
yep, these changes are just to remove the (many) warnings from pyo3